### PR TITLE
refactor: moving consul access and role interface

### PR DESCRIPTION
**Pull request by ItsRico**
Moving configureConsulAccess and createRole to go-mod-secrets to consolidate consul API.

Closes: #3227

Signed-off-by: Rico Chavez-Lopez <rchavezlopez@ucdavis.edu>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x ] I am not introducing a new dependency (add notes below if you are)
- [x ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

https://github.com/ItsRico/edgex-go/tree/3227-testing
To test these changes,

1. Checkout this branch
2. Checkout https://github.com/edgexfoundry/go-mod-secrets/pull/163 in a subfolder of this branch
3. The following two steps can be viewed here: https://github.com/ItsRico/edgex-go/compare/main...ItsRico:edgex-go:3227-testing?expand=1
4. Update your go.mod file using the replace directive. ( => ./go-mod-secrets)
5. Update your Makefile to copy go-mod-secrets (add a COPY to the inline Dockerfile)
 In edgex-go, run
7. make test
8. make clean
9. make clean_docker_base
10. make docker_base
11. make dev
12. Then, in your edgex-compose/compose-builder/ folder, 
13. make run dev

The unit tests for these changes can be found in aclroles_test.go and in go-mod-secrets's management_test.go.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -42,8 +42,6 @@ github.com/edgexfoundry/go-mod-messaging/v2 v2.3.0-dev.15 h1:Kddm7MEqId6j3sXVaA5
 github.com/edgexfoundry/go-mod-messaging/v2 v2.3.0-dev.15/go.mod h1:yLJ9EK4Feg409FDr0oP87LbaRLyOSGJk/ikaIfEDKcI=
 github.com/edgexfoundry/go-mod-registry/v2 v2.2.0 h1:dk9ul1t7INAiyZXeu/GrpinFE3qOekdy8uZOqEGgIiE=
 github.com/edgexfoundry/go-mod-registry/v2 v2.2.0/go.mod h1:DUQRnAd5fVzoROc5SI+PTFUD/vCNeZmZHBMrLElbmwI=
-github.com/edgexfoundry/go-mod-secrets/v2 v2.3.0-dev.7 h1:BFg7lzSVitQGvGY1mZucMXI5HNLgORDWde8a3L5dWsM=
-github.com/edgexfoundry/go-mod-secrets/v2 v2.3.0-dev.7/go.mod h1:h/FohFNY8xHalioLg1bhjAuEj0z+danSDtixirvaXmQ=
```
Comment (jim-wang-intel) [inline go.sum:46]: we still need go-mod-secrets but different version i think

File: internal/security/bootstrapper/command/setupacl/aclroles_test.go [hunk 2]
```diff
@@ -17,27 +17,29 @@ package setupacl
 
 import (
 	"context"
+	"errors"
```
Comment (jim-wang-intel) [inline internal/security/bootstrapper/command/setupacl/aclroles_test.go:20]: should move this test to go-mod-secret?

File: internal/security/bootstrapper/command/setupacl/command.go [hunk 3]
```diff
@@ -329,20 +342,89 @@ func (c *cmd) createEdgeXACLTokenRoles(bootstrapACLTokenID, secretstoreToken str
...
 			return fmt.Errorf("failed to create edgex role: %v", err)
 		}
 	}
 
 	return nil
 }
 
+// configureConsulAccess is to enable the Consul config access to the SecretStore via consul/config/access API
+// see the reference: https://www.vaultproject.io/api-docs/secret/consul#configure-access
+func (c *cmd) configureConsulAccess(secretStoreToken string, bootstrapACLToken string) error {
```
Comment (jim-wang-intel) [inline internal/security/bootstrapper/command/setupacl/command.go:371]: i thought this func already moved to go.mod.secrets?

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
still some questions and should refactor more.

File: internal/security/bootstrapper/command/setupacl/command.go (hunk 3 above)
Comment (ItsRico) [reply internal/security/bootstrapper/command/setupacl/command.go:371]: My mistake, will remove

File: internal/security/bootstrapper/command/setupacl/aclroles_test.go (hunk 2 above)
Comment (bnevis-i) [reply internal/security/bootstrapper/command/setupacl/aclroles_test.go:20]: Agree.  This should have gone along with aclroles.go.

File: go.sum (hunk 1 above)
Comment (ItsRico) [reply go.sum]: Reverted

**Comment by ItsRico** [comment]
- removed aclroles_test.go

Still need to add helper function for client creation and remove from loop

**Comment by ItsRico** [comment]
- Updated reference to NewRegistryRole to NewConsulRole, reflecting corresponding changes in go-mod-secrets
- Moved client creation outside of for loop for CreateRole
- Reran Testing Instructions

image

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by ItsRico** [comment]
Bumped version for go-mod-secrets, reran testing, rebasing

**Comment by ItsRico** [comment]
Had an issue with the rebase, need to try again

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by bnevis-i — APPROVED** [review]

**Pull request merged**